### ISSUE TITLE
ValueError: Did not find PCA in `.obsm.keys()

### ISSUE THREAD

**Issue by MinjieHu** [zhanghao-njmu/SCP#144]
Hi there,
Thanks for the great pipeline!

I encountered this error when I try to run RunSCVELO. Here is error message
```
Warning: sparse->dense coercion: allocating vector of size 1.1 GiBAssay 'integrated' is in the srt object but not converted.
Assay 'ambiguous' is in the srt object but not converted.
'misc' slot is not converted.
'tools' slot is not converted.
posx and posy should be finite values
posx and posy should be finite values
posx and posy should be finite values
posx and posy should be finite values
Filtered out 13400 genes that are detected 30 counts (shared).
Normalized count data: X, spliced, unspliced.
WARNING: Did not modify X as it looks preprocessed already.
computing neighbors
Error: ValueError: Did not find PCA in `.obsm.keys()`. You need to compute it first.
```

I have tried to re-run Standard_SCP or RunPCA, but neither works.

Thanks in advance for the help!

Minjie

**Comment by MinjieHu** [comment]
When I tried to run the example data, I run into an error "Error: AttributeError: can't set attribute"

Here is all the code I used
```
data("pancreas_sub")
pancreas_sub <- Standard_SCP(srt = pancreas_sub)
pancreas_sub <- RunSCVELO(
  srt = pancreas_sub, group_by = "SubCellType",
  linear_reduction = "PCA", nonlinear_reduction = "UMAP"
)
VelocityPlot(srt = pancreas_sub, reduction = "UMAP", group_by = "SubCellType")
```

Hope I can get more clues about why it happens.

**Comment by MinjieHu** [comment]
I got the first issue solved. The name for PCA reduction in my object is “pca”. When using “PCA” indicated in the code doesn’t match with the reduction name in the object.

However, after getting it solved, I ran into the same error as the example data

**Comment by zhanghao-njmu** [comment]
The issue was caused by the "srt_to_adata" function. SCP has fixed this bug in version v0.5.1. You can try installing this version and running the tests again.

**Comment by jjccco** [comment]
> I got the first issue solved. The name for PCA reduction in my object is “pca”. When using “PCA” indicated in the code doesn’t match with the reduction name in the object.
> 
> However, after getting it solved, I ran into the same error as the example data

I just solve this question by using lowercase, like "pca" and "umap".
